xapian-core/cluster/cluster.cc

```diff
@@ -121,10 +85,11 @@ TermListGroup::add_document(const Document &document) {
 }
 
 void
-TermListGroup::add_documents(MSetDocumentSource docs) {
+TermListGroup::add_documents(const MSet &docs) {
     LOGCALL_VOID(API, "TermListGroup::add_documents()", docs);
-    while (!docs.at_end())
-	add_document(docs.next_document());
+    MSetIterator it = docs.begin();
```

Comment:
You can put this initialiser inside the `for`. This would be more natural C++. See for example the code in the user guide. (Unless you build it yourself, it's probably easiest to look at the source code to get the C++ examples.)